src/test/java/com/amazon/ion/impl/IonReaderContinuableTopLevelBinaryTest.java

```diff
@@ -1666,11 +1666,15 @@ public void byteSizeNotOnLobFails(boolean constructFromBytes) throws Exception {
 
     @ParameterizedTest(name = "constructFromBytes={0}")
     @ValueSource(booleans = {true, false})
-    public void doubleValueOnIntFails(boolean constructFromBytes) throws Exception {
-        reader = readerFor(constructFromBytes, 0x20);
-        reader.next();
-        assertThrows(IllegalStateException.class, () -> reader.doubleValue());
-        reader.close();
+    public void doubleValueOnInt(boolean constructFromBytes) throws Exception {
+        reader = readerFor(constructFromBytes, 0x20, 0x31, 0x1, 0x21, 0xFF);
+        assertSequence(
+            next(IonType.INT), doubleValue(0.0),
+            next(IonType.INT), doubleValue(-1.0),
+            next(IonType.INT), doubleValue(255.0),
+            next(null)
+        );
+        closeAndCount();
     }
```

Comment:
Whether we support coercing arbitrary size Ints or not we should have a test for that.

Reply:
Added